x-pack/metricbeat/module/oracle/tablespace/data.go

```diff
@@ -41,13 +41,14 @@ func (m *MetricSet) transform(in *extractedData) (out map[string]mapstr.M) {
 	out = make(map[string]mapstr.M, 0)
 
 	for _, dataFile := range in.dataFiles {
-		m.addDataFileData(&dataFile, out)
+		dataFileCopy := dataFile
+		m.addDataFileData(&dataFileCopy, out)
```

Comment:
```suggestion
		dataFileCopy := dataFile
		m.addDataFileData(&dataFileCopy, out)
```

This gotcha is no longer valid starting from Go 1.22 and above. See the first point of this [link](https://go.dev/ref/spec#Go_1.22) related to "for". Currently, for beats, we are using Go 1.21 but I am pretty sure we will move to 1.22 very soon.

So, for now, we can fix it. But after Go 1.22 this is no more required. Also, if you look deeper into how that dataFile is used; it is just being read and not written to and that's why we never faced an issue where passing the address of the var caused us an issue here.

A cleaner way to address it for now would be to do this (without creating a new var):

```go
for i := range in.dataFiles {
	m.addDataFileData(&in.dataFiles[i], out)
}
```

Reply:
@niraj-elastic made this change to make the linter happy. So, either we have to put a comment of nolint or make the change to keep the linter happy.

I am more inclined towards making the following change as suppose next month we are still in Go 1.21 and someone changes the code such that dataFiles is even being modified but we won't be able to get any lint errors as we have put a nolint comment.

```
for i := range in.dataFiles {
	m.addDataFileData(&in.dataFiles[i], out)
}
```

Reply:
I have updated the code with suggested changes.